pkg/spec/spec.go

```
@@ -73,6 +74,11 @@ func CreateConfigToOCISpec(config *CreateConfig) (*spec.Spec, error) { //nolint
 		g.AddAnnotation(key, val)
 	}
 	g.SetRootReadonly(config.ReadOnlyRootfs)
+	if config.Hostname == "" {
+		if config.NetMode.IsHost() {
+			config.Hostname, _ = os.Hostname()
```

Shouldn't discard this error

---

Fixed